index.html

```diff
@@ -811,6 +814,120 @@ <h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest" id="identity">id</dfn></code> member is a <a>string</a> that
+          represents the <dfn>identity</dfn> for the application. 
+          The [=identity=] takes the form of a URL, which is same origin as the
+          start URL.
+        </p>
+        <p>The [=identity=] is used by user agents to uniquely identify the
+        application universally. When the user agent sees a manifest with an
+        [=identity=] that does not correspond to an already-installed application,
+        it SHOULD treat that manifest as a description of a distinct
+        application, even if it is served from the same URL as that of another
+        application. When the user agent sees a manifest with an [=identity=]
+        matching that of an already-installed application, it SHOULD be used as
+        a signal that this manifest is a replacement for the already-installed
+        application's manifest, and not a distinct application, even if it is
+        served from a different URL than the one seen previously.
+        </p>
+        <p>
+        The [=identity=] is considered matching when the processed |manifest|["id"] [=url/equal|equals=].
+        </p>
+        <p class="note">
+          The [=identity=] can be used by a service that collects lists of web
+          applications to uniquely identify applications.
+        </p>
+        <p class="note">
+          The [=identity=] is processed like a URL but it doesn't point to a
+          resource that can be navigated to, so it's not required to be
+          [=URL/within scope=].
+        </p>
+        <p>
+          To <dfn>process the `id` member</dfn>, given [=object=]
+          |json:JSON|, [=ordered map=] |manifest:ordered map|:
+        </p>
+        <ol class="algorithm">
+          <li>Set |manifest|["id"] to |manifest|["start_url"].
+          </li>
+          <li>If the type of |json|["id"] is not [=string=], return.
+          </li>
+          <li>Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"]
+          with |manifest|["start_url"]'s origin as the base URL.
+          </li>
+          <li>If |id| is failure, return.
+          </li>
+          <li>If |id| is not [=same origin=] as |manifest|["start_url"], return.
+          </li>
+          <li>Otherwise, set |manifest|["id"] to |id|.</li>
+        </ol>
+        <aside class="example">
+          <p>Below table shows some example cases of the |identity| processing algorithm.</p>
+          <table class="data">
+            <tr>
+              <th>|json|["id"]</th>
+              <th>|manifest|["start_url"]</th>
+              <th>|manifest|["id"]</th>
+            </tr>
+            <tr>
+              <td><i>undefined</i></td>
+              <td>"https://example.com/my-app/start"</td>
+              <td>"https://example.com/my-app/start"</td>
+            </tr>
+            <tr>
+              <td>""</td>
```

Comment:
Hmm... I don't know about this case... I think we should treat the empty string as missing.

Reply:
We do the same for start_url, I think (i.e., treat empty string as whatever the default is).

Reply:
I don't have a strong preference here. @dmurph @mgiuca WDYT?

Reply:
I think I have a weak preference to agree with @marcoscaceres and treat it as missing (fallback to `start_url`).

Overall I think we've gone down a pretty confusing path with this being resolved against the origin but defaulting to `start_url`.

Not sure if it's too late to reconsider, but could we actually **resolve it against `start_url`**.

Pros:

* It just means if it's empty, or missing, it would naturally be equal to `start_url`, without needing a special case for "if it's empty".
* It makes it much more important to add a leading '/' to the `id`, to anchor it to the origin. Which is something I've been, on a few other threads, saying we should be recommending anyway, to really make it clear that this is a URL and the '/' makes it unambiguous that it is relative to the origin.

Cons:

* It means if you don't put a leading '/' on the `id`, if you change the `start_url` your identity will change, which destroys the whole point of having `id`.

So that Con doesn't sound great, right? But if we go with "`""` means `start_url`", that Con also applies, in that one special case — if you want the id to be empty you need to remember to write `"/"` rather than `""`. So maybe my suggestion makes sense and it's just quite important to prefix your `id` with `/`.

Reply:
If we aren't keen on my above suggestion (a quick chat with Phillis offline suggests not), then I think going with Marcos' suggestion is good. It's actually consistent with how [scope is parsed](https://www.w3.org/TR/appmanifest/#dfn-process-the-scope-member):

- If it's missing, invalid, **or empty**, it defaults to `start_url`.
- If not, it's parsed against manifest URL.

So we've already got that weird condition "if empty default to something other than the base URL it's being parsed against" in the `scope` processor. We would be doing the same thing here.

Ultimately, if we recommend that users always start `id` with '/', then none of this matters, so let's make sure we always recommend that and do this small change and move on.

Reply:
thanks Matt! Yeah my main concern is with the con you already said. If we make start_url as the base_url, then I think the leading slash has to be required, which then basically mean the base_url is the root. Otherwise using start_url as base_url + not requiring to have leading slash make it very error-prone to devs.

I will make the update for the empty string!